### PR TITLE
added book entity , book repo

### PR DESCRIPTION
**Pull request by chertterner**

### CODE REVIEW

File: src/main/java/com/example/bookshop/BookShopApplication.java [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+            book.setDescription(" ");
+            book.setIsbn(" ");
+            book.setPrice(BigDecimal.ONE);
+            book.setTitle(" ");
+            book.setAuthor(" ");
+
+            bookService.save(book);
+            System.out.println(bookService.findAll());
+        };
+    }
```
Comment (sokimaaa) [inline src/main/java/com/example/bookshop/BookShopApplication.java:34]:
```suggestion
 
    public static void main(String[] args) {
        SpringApplication.run(BookShopApplication.class, args);
    }

    @Bean
    public CommandLineRunner commandLineRunner(BookService bookService) {
        return args -> {
            Book book = new Book();
            book.setDescription(" ");
            book.setIsbn(" ");
            book.setPrice(BigDecimal.ONE);
            book.setTitle(" ");
            book.setAuthor(" ");

            bookService.save(book);
            System.out.println(bookService.findAll());
        };
    }
```

File: src/main/java/com/example/bookshop/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (sokimaaa) [inline src/main/java/com/example/bookshop/model/Book.java]: Using @Data in conjunction with @Entity is not recommended.

File: src/main/java/com/example/bookshop/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
```
Comment (sokimaaa) [inline src/main/java/com/example/bookshop/model/Book.java]: I see any constraints requested by the task, e.g. not null, unique

File: src/main/java/com/example/bookshop/repository/implementation/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    @Transactional
```
Comment (sokimaaa) [inline src/main/java/com/example/bookshop/repository/implementation/BookRepositoryImpl.java]: use one of the methods to reach the transaction (not both!).
1) using @Transactional
2) using SessionFactory#beginTransaction and its methods

**Review by aaananas — APPROVED** [review]

**Review by koval-kn — CHANGES_REQUESTED** [review]
java ci checks failed

**Pull request merged**